zerver/webhooks/ansibletower/view.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, Dict, Iterable, Optional, List
+
+from django.utils.translation import ugettext as _
+from django.http import HttpRequest, HttpResponse
+
+from zerver.lib.webhooks.common import check_send_webhook_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.models import UserProfile
+
+import ujson
+import operator
+
+ANSIBLETOWER_SUBJECT_TEMPLATE = "{name}"
+
+ANSIBLETOWER_DEFAULT_MESSAGE_TEMPLATE = ("Job Type: {friendly_name}\n"
+                                         "ID: [{id}]({url})\n"
+                                         "Name: {name}\n"
+                                         "Status: {status}")
+
+ANSIBLETOWER_JOB_MESSAGE_TEMPLATE = ("Job Type: {friendly_name}\n"
+                                     "ID: [{id}]({url})\n"
+                                     "Name: {name}\n"
+                                     "Status: {status}\n"
+                                     "Hosts:\n"
+                                     "{hosts_final_data}")
+
+ANSIBLETOWER_JOB_HOST_ROW_TEMPLATE = '* {hostname}: {status}\n'
+
+@api_key_only_webhook_view('Ansibletower')
+@has_request_variables
+def api_ansibletower_webhook(request: HttpRequest, user_profile: UserProfile,
+                             payload: Dict[str, Any]=REQ(argument_type='body')) -> HttpResponse:
+
+    body = get_body(payload)
+    subject = get_subject(payload)
+
+    check_send_webhook_message(request, user_profile, subject, body)
+    return json_success()
+
+def get_subject(payload: Dict[str, Any]) -> str:
+    return ANSIBLETOWER_SUBJECT_TEMPLATE.format(name=payload['name'])
+
+def get_body(payload: Dict[str, Any]) -> str:
+    if (payload['friendly_name'] == 'Job'):
+        hosts_list_data = payload['hosts']
+        hosts_data = []
+        for host in payload['hosts']:
+            if (hosts_list_data[host].get('failed') is True):
+                hoststatus = 'Failed'
+            elif (hosts_list_data[host].get('failed') is False):
+                hoststatus = 'Success'
+            hosts_data.append({
+                'hostname': host,
+                'status': hoststatus
+            })
+
+        name = payload['name']
+        friendly_name = payload['friendly_name']
+        job_id = payload['id']
+        url = payload['url']
+        status = payload['status']
```

Comment:
You don't need placeholder variables here, you can just pass in stuff like `payload['name']` directly to the `format()` call below.